src/libstd/sys/common/net2.rs

```diff
@@ -268,6 +269,16 @@ impl FromInner<Socket> for TcpStream {
     }
 }
 
+impl fmt::Debug for TcpStream {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("TcpStream")
+        .field("addr", &self.socket_addr())
+        .field("peer", &self.peer_addr())
+        .field("inner", &self.inner.as_inner())
```

Comment:
It may be worth using using the windows/unix specific naming here to say whether it's a file descriptor or a socket. For example something like:

```rust
let name = if cfg!(windows) {"socket"} else {"fd"};
// ...
    .field(name, &self.inner.as_inner())
```